blockchain/types/tx_internal_data_value_transfer_memo.go

```diff
@@ -315,7 +316,11 @@ func (t *TxInternalDataValueTransferMemo) Validate(stateDB StateDB, currentBlock
 }
 
 func (t *TxInternalDataValueTransferMemo) ValidateMutableValue(stateDB StateDB, currentBlockNumber uint64) error {
-	if stateDB.IsProgramAccount(t.Recipient) {
+	acc := stateDB.GetAccount(t.Recipient)
```

Comment:
Just use `IsContractAccount`

Reply:
Syntax-wise you're right. But we're going to modify the validation rule for those transaction types. Please see "Interaction with other transaction types" and "Kaia transaction type restrictions" sections of https://github.com/blukat29/kips/blob/setcode/KIPs/kip-xxx.md

Reply:
Thanks for your information. These specs help me with my understanding.

Reply:
I check the KIP and change it for each transaction type.

Reply:
Shouldn't the condition be as follows?
```
if acc.Type() is SCAType OR acc code is not empty code hash {
	return kerrors.ErrNotForProgramAccount
}
```

Reply:
Added a function to check the account state for tx according to KIP.

https://github.com/kaiachain/kaia/pull/126/files#diff-36960dea371fb7a7722f48b0905c890030f61801a29c132b5ade154357f0e1dbR681